### DISCUSSION TITLE
Possible solution to binding multiple RX with multiple TX (and simulteously)

### DISCUSSION THREAD

**Discussion by warshoot3r** [ExpressLRS/ExpressLRS#2652] [Ideas]
### Background
I manage a collection of drones and ELRS (ExpressLRS) equipment, including multiple radios and TX modules. Currently, I encounter difficulties in maintaining a simple, efficient binding process across multiple drones and radios, particularly when sharing equipment with others or managing a small drone fleet.

### Current Problem
The primary issue arises when using the same binding phrase for all ELRS receivers (RX) and transmitters (TX). This setup works well when flying solo but leads to binding conflicts when multiple people use different radios simultaneously. Here are some current challenges:

- Complexity for Simultaneous Flying: Methods to enable simultaneous flying are cumbersome and not practical in field conditions.

- Dependence on External Devices: Most workarounds require a smartphone or computer.

- Impact on Solo Flying: Alterations needed for group flying complicate the simple "turn it on" method when flying alone, potentially causing drones to unbind.

- Management Overhead: These workarounds increase the complexity of managing the ELRS setup.

### Current Workarounds

Several existing methods attempt to address this issue, but each introduces its own set of complications:

1. Separate Binding Phrases: Requires frequent reconfiguration of RX and TX, which is time-consuming.
2. Manual Binding without a Phrase: Involves repeated binding procedures, which are inefficient in the long run.
3. Module-Specific Binding Phrases: Necessitates carrying specific ELRS modules, not ideal for built-in transmitters.
4. Use of SPI ELRS with App Configuration: Imposes additional costs and requires software reconfiguration.
5. Changing RX/TX Configurations Per Flight: Similar to using separate phrases, requires constant attention to settings.

### Proposed Feature: "Guest Mode"
To address these challenges while maintaining the simplicity and effectiveness of the ELRS system, I propose a new feature for ELRS TX modules called "Guest Mode." This feature would function as follows:

> Activation: Enable via the ELRS LUA script.
> 
> Function: Automatically assigns a random or preset, non-user-configurable binding phrase temporarily.
> 
> Use Case: Facilitates the temporary binding of drones to different transmitters, ideal for scenarios where multiple users need to operate different drones without permanent reconfiguration.
> 

### Implementation Steps for "Guest Mode"

Pilot 1 (Owner):

> Selects drones to lend.
> Powers on their transmitter and drone as usual.
> Activates the "loan model" feature on the drone.
> 

Pilot 2 (Guest):

> Powers on their radio.
> Activates "Guest Mode" in the ELRS LUA script.
> Initiates binding. The drone binds using the temporary phrase from "Guest Mode."

Pilot 1 (Owner):

> Operates their own setup as usual with no additional configuration needed.
> 
> Additionally pilot 1 will perform the 3 power on/off to return the drone to the original bindings.
> 

### Summary
This feature would allow the owner to use any combination of ELRS equipment when flying solo without concerns about binding issues. When flying in a group, "Guest Mode" simplifies equipment sharing without requiring constant reconfiguration, thus preserving the ease of use and flexibility of the ELRS system.

**Comment by JyeSmith** [comment]
Test the current release candidate for v3.5 that came out a couple of days ago.  There is a rx Lua setting to make the bind volatile.  This means that every time the rx powers up it will be in bind mode.  Flash each tx with a different bind phrase, then press bind and fly. 

You can even configure module buttons (like on the Radiomaster Ranger/Bandit) for binding.  This makes it a little quicker than entering Lua.

**Reply by pkendall64** [reply]
That would be 3.4RC1 NOT 3.5 😄
